docs/core/porting/upgrade-assistant-uwp-framework.md

```diff
@@ -0,0 +1,524 @@
+author: adegeo
+ms.date: 05/17/2022
+---
+# Upgrade a UWP App to .NET 6 and WinUI3 with Windows App SDK capabilities using the .NET Upgrade Assistant
+
+The [.NET Upgrade Assistant](upgrade-assistant-overview.md) is a command-line tool that can assist with upgrading .NET Framework WPF apps to .NET 6. This article provides:
+
+- What to expect from the tool?
+- Things to know before starting your migration journey 
+- A demonstration of how to run the tool against a UWP app
+- Troubleshooting tips
+
+For more information on how to install the tool, see [Overview of the .NET Upgrade Assistant](upgrade-assistant-overview.md).
+
+## What to expect from the tool?
+
+We’re working on an update to the .NET Upgrade Assistant which is an open-source tool to help you automate migration of .NET Native UWP apps to .NET 6, including updating from WinUI 2 to WinUI 3. This is a command line tool that leverages .NET Upgrade assistant and will live as an extension in the same github repo [here](https://github.com/dotnet/upgrade-assistant.git). This tool aims to provide a simple migration experience by 
+
+- Backing up your UWP project 
+- Convert project file (csproj) to SDK style
+- Clean up NuGet package references 
+- Update the target framework and NuGet packages 
+- Update package.appxmanifest
+- Removal of properties and items that are no longer required
+- Add template files such as App.Xaml, MainWindow.Xaml and publish profiles with options to build packaged and unpackaged app
+- Make namespace changes and add MainPage navigation
+- An attempt to detect and fix apis that have changed from UWP to Windows App SDK
+- An attempt to detect and mark apis that are either no longer supported or cannot be automatically converted with TODO comments and CLI messages
+
+We aim to provide migration guidance in form of warning messages within the tool and TODO comments within your project as the tool tries to migrate the project. In this way, you’ll always be in control of your migration. And for the APIs where complete automation is not possible, plan is to add //TODO comments for the developers to know where the work will be needed. A typical //TODO comment will also include a link to our existing migration documentation. Please check the Task list within the Visual Studio to see all the action items as TODO comments.
+
+## Things to know before starting your migration journey
+
+This tool currently supports C# and in most cases the app will require additional effort to complete the migration. The goal of the tool will be to convert your project and code so that it can compile, but features that require additional developer investigation will have TODO comments. Please check out the what’s supported page [here] (https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/what-is-supported)] before you begin your migration journey.
```

Comment:
Check all of your links. Don't use a locale: `en-us` and if the link is to the docs website, simply start at the root without domain `/url/path/to/article`.

Also, don't say "please" as you're not begging readers to do something :D and instead just tell them what to do. Also, be more direct with your wording, don't use `it will do x` just say `It does x`. We also use the standardized "For more information" syntax when telling someone to look at extra info.

Please look at all of your URLs and do the same changes. Here are all those fixes in a single commit for you:

```suggestion
This tool currently supports C#, and in most cases the app will require additional effort to complete the migration. The goal of the tool is to convert your project and code, so that it can compile. Some features require you to investigate and fix, and have `//TODO` code comments. For more information about what to consider before migrating, see [What is supported when migrating from UWP to WinUI 3](/windows/apps/windows-app-sdk/migrate-to-windows-app-sdk/what-is-supported).
```